### ISSUE TITLE
simplifying trigonometric 

### ISSUE THREAD

**Issue by silveiralucas** [sympy/sympy#16705]
I can't get the simplify function to simplify the trigonometric identities when integrals of functions are present. I put a simple example where it does not work as expected.

from IPython.display import display
from sympy import *

# Symbolic constants
z = symbols('z')
R = symbols('R')
theta = symbols('theta')

# Functions
phi_x = Function('\phi_x')(z)
phi_y = Function('\phi_y')(z)
m = Function('m')(z)

expr = sin(theta)**2*integrate(phi_y**2*m, (z, 0, R)) + cos(theta)**2*integrate(phi_y**2*m, (z, 0, R)) + integrate(phi_x**2*m, (z, 0, R))
expr = simplify(expr)
display(expr)

----------------------------------------------------------------------------------------------

[out]:sin(theta)**2*Integral(\phi_y(z)**2*m(z), (z, 0, R)) + cos(theta)**2*Integral(\phi_y(z)**2*m(z), (z, 0, R)) + Integral(\phi_x(z)**2*m(z), (z, 0, R))

[expected]: Integral(\phi_y(z)**2*m(z), (z, 0, R)) + Integral(\phi_x(z)**2*m(z), (z, 0, R))

**Comment by oscarbenjamin** [comment]
Here's a simpler example:
```julia
In [47]: sin(x)**2*Integral(y, y) + cos(x)**2*Integral(y, y) + z                                                                                              
Out[47]: 
       2    ⌠           2    ⌠     
z + sin (x)⋅⎮ y dy + cos (x)⋅⎮ y dy
            ⌡                ⌡     

In [48]: simplify(_)                                                                                                                                          
Out[48]: 
       2    ⌠           2    ⌠     
z + sin (x)⋅⎮ y dy + cos (x)⋅⎮ y dy
            ⌡                ⌡     

In [49]: factor(_)                                                                                                                                            
Out[49]: 
       2    ⌠           2    ⌠     
z + sin (x)⋅⎮ y dy + cos (x)⋅⎮ y dy
            ⌡                ⌡     
```
Maybe it is related to the inability to factor expressions like `a*b + a*c + d`
```julia
In [51]: factor(x*z + x*y + t)                                                                                                                                
Out[51]: t + x⋅y + x⋅z
```
since it works fine without the additional term:
```julia
In [54]: sin(x)**2*Integral(y, y) + cos(x)**2*Integral(y, y)                                                                                                  
Out[54]: 
   2    ⌠           2    ⌠     
sin (x)⋅⎮ y dy + cos (x)⋅⎮ y dy
        ⌡                ⌡     

In [55]: simplify(_)                                                                                                                                          
Out[55]: 
⌠     
⎮ y dy
⌡ 
```

Although the additional term doesn't matter if the factor is a symbol instead of an integral:
```julia
In [52]: sin(x)**2*y + cos(x)**2*y + z                                                                                                                        
Out[52]: 
     2           2       
y⋅sin (x) + y⋅cos (x) + z

In [53]: simplify(_)                                                                                                                                          
Out[53]: y + z
```

**Comment by silveiralucas** [comment]
I agree it is most likely related to the inability to factor when there is a third term. However I get the correct solution when there is no integral in the terms containing the trigonometric functions. Even with an integral on the third term, it outputs the correct solution.

# Symbolic constants
z = symbols('z')
R = symbols('R')
theta = symbols('theta')

# Functions
phi_x = Function('\phi_x')(z)
phi_y = Function('\phi_y')(z)
m = Function('m')(z)

#%%

expr = sin(theta)**2*phi_y**2*m + cos(theta)**2*phi_y**2*m + integrate(phi_x**2*m, (z, 0, R))
expr = simplify(expr)
display(expr)

[out] = \phi_y(z)**2*m(z) + Integral(\phi_x(z)**2*m(z), (z, 0, R))

**Comment by oscarbenjamin** [comment]
I edited your original comment to fix the markdown problems can you do the same to your last comment (note how all the code has come out funny). You should surround code in triple backtick blocks:
https://help.github.com/en/articles/creating-and-highlighting-code-blocks

You can use the preview tab to see what your comment will look like before finishing it.

**Comment by silveiralucas** [comment]
Thanks. Any clue on how to solve the issue? My code has long equations with many symbolic integrals and trigonometric identities.
